contracts/Everlong.sol

```diff
@@ -62,16 +66,58 @@ import { EverlongPositions } from "./internal/EverlongPositions.sol";
 /// @custom:disclaimer The language used in this code is for coding convenience
 ///                    only, and is not intended to, and does not, have any
 ///                    particular legal or regulatory significance.
-contract Everlong is EverlongAdmin, EverlongPositions {
-    /// @notice Initial configuration paramters for Everlong.
-    /// @param _name Name of the ERC20 token managed by Everlong.
-    /// @param _symbol Symbol of the ERC20 token managed by Everlong.
-    /// @param __hyperdrive Address of the Hyperdrive instance wrapped by Everlong.
-    /// @param __asBase Whether to use Hyperdrive's base token for bond purchases.
+contract Everlong is EverlongAdmin, EverlongPositions, EverlongERC4626 {
+    // ╭─────────────────────────────────────────────────────────╮
+    // │ Constructor                                             │
+    // ╰─────────────────────────────────────────────────────────╯
+
+    /// @notice Initial configuration paramters for EverlongERC4626.
+    /// @param __name Name of the ERC20 token managed by Everlong.
+    /// @param __symbol Symbol of the ERC20 token managed by Everlong.
+    /// @param __hyperdrive Address of the Hyperdrive instance.
+    /// @param __asBase Whether to use the base or shares token from Hyperdrive.
     constructor(
-        string memory _name,
-        string memory _symbol,
+        string memory __name,
+        string memory __symbol,
         address __hyperdrive,
         bool __asBase
-    ) EverlongBase(_name, _symbol, __hyperdrive, __asBase) {}
+    ) {
+        // Store constructor parameters.
+        _name = __name;
+        _symbol = __symbol;
+        _hyperdrive = __hyperdrive;
+        _asBase = __asBase;
+        _asset = __asBase
+            ? IHyperdrive(__hyperdrive).baseToken()
+            : IHyperdrive(__hyperdrive).vaultSharesToken();
+
+        // Set the admin to the contract deployer.
+        _admin = msg.sender;
+
+        // Attempt to retrieve the decimals from the {_asset} contract.
+        // If it does not implement `decimals() (uint256)`, use the default.
```

Comment:
Thinking about yield sources like `ChainlinkHyperdrive` or other kinds of bridged assets, it may be better to just have the deployer provide a `decimals` parameter.

Reply:
Totally agree, good callout